pod2daemon/Makefile

```
@@ -127,12 +127,15 @@ $(BINDIR)/node-driver-registrar-%: clone-registrar-upstream
 clone-registrar-upstream:
 ifeq ("$(wildcard ./$(REGISTRAR_IMAGE))", "")
 	@echo "Directory does not exist."
-	git clone --depth 1 --branch $(UPSTREAM_REGISTRAR_TAG) --single-branch https://github.com/$(UPSTREAM_REGISTRAR_PROJECT).git
-	patch -d node-driver-registrar -p1 < patches/0001-Update-go.mod-to-patch-CVEs.patch
-	rm -rf node-driver-registrar/vendor
+	git clone --depth 1 --single-branch https://github.com/$(UPSTREAM_REGISTRAR_PROJECT).git
 else
 	@echo "Upstream repo already cloned."
 endif
+	cd $(REGISTRAR_IMAGE) && \
+	  git checkout $(UPSTREAM_REGISTRAR_TAG) || \
+	  git fetch --tags origin $(UPSTREAM_REGISTRAR_TAG) && \
+	  git checkout $(UPSTREAM_REGISTRAR_TAG)
+	rm -rf ./$(REGISTRAR_IMAGE)/vendor
 
 clean-registrar:
 	rm -rf node-driver-registrar
```

not at all necessary, but it would be nice:

```suggestion
	rm -rf ./$(REGISTRAR_IMAGE)
```